### DISCUSSION TITLE
Not installing dependencies inside npm packages / next js node react

### DISCUSSION THREAD

**Discussion by romanusname** [vercel/next.js#46494] [Help]
I ran into a very strange problem. I'm surprised there isn't a solution on the internet.
I install npm packages which have their dependencies inside.
These dependencies do not install themselves.

I am getting a lot of warnings in the console. For example:

```
Module not found: Can't resolve 'snappy'
Module not found: Can't resolve 'bufferutil'
Module not found: Can't resolve 'mongodb-client-encryption'
Module not found: Can't resolve 'aws4'
```
These dependencies are not required, but they pollute the console a lot!
All of the above are mongoose dependencies

I tried:
```
npm install -g mongoose
npm install mongoose --save
```
no results.

The most terrible! I tried installing these dependencies manually. There are about 10 of them, but I decided to make it.
**EVERY DEPENDENCE INSIDE SOYA DEPENDENCE!** They also do not install automatically, and clog the console even more!

Please help fix this. I wasted a lot of time :(
**I am using next js latest version**

**Comment by icyJoseph** [comment]
Can you share a repository or stackblitz demo? I think NPM doesn't install peer dependencies by default? or something like that? what npm version have you got?

**Reply by romanusname** [reply]
I just reinstalled node js and npm.
npm version - 9.5.0
LTS Recommended For Most Users
